### PR TITLE
Support circuit breaker group name

### PR DESCRIPTION
**Pull request by Ferioney**
Fixes gh-514

### CODE REVIEW

**Comment by Ferioney** [comment]
This changes required the latest version of Spring Cloud Commons

File: spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/FeignCircuitBreakerInvocationHandler.java [hunk 1]
```diff
@@ -76,7 +79,7 @@ else if ("toString".equals(method.getName())) {
 			return toString();
 		}
 		String circuitName = Feign.configKey(target.type(), method);
-		CircuitBreaker circuitBreaker = this.factory.create(circuitName);
+		CircuitBreaker circuitBreaker = this.factory.create(circuitName, this.feignClientName);
```
Comment (OlgaMaciaszek) [inline spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/FeignCircuitBreakerInvocationHandler.java:82]: Please remove unnecessary `this.` keyword.

**Review by OlgaMaciaszek — CHANGES_REQUESTED** [review]
@Ferioney Thanks for submitting the PR. Apart from a cosmetic change (see comment), this generally looks good. However, as it would be changing existing behaviour and, thus, potentially breaking backwards compatibility for some users, it will need to be changed to an opt-in feature (for example, using a property flag), and then changed into the default behaviour in a major release.

File: spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/FeignCircuitBreakerInvocationHandler.java (hunk 1 above)
Comment (Ferioney) [reply spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/FeignCircuitBreakerInvocationHandler.java]: Removed

File: spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/FeignCircuitBreakerTargeter.java [hunk 2]
```diff
@@ -81,7 +84,8 @@
 	}
 
 	private FeignCircuitBreaker.Builder builder(String feignClientName, FeignCircuitBreaker.Builder builder) {
-		return builder.circuitBreakerFactory(this.circuitBreakerFactory).feignClientName(feignClientName);
+		return builder.circuitBreakerFactory(this.circuitBreakerFactory).feignClientName(feignClientName)
```
Comment (OlgaMaciaszek) [inline spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/FeignCircuitBreakerTargeter.java:87]: Please remove unnecessary `this.` keywords.

**Review by OlgaMaciaszek — CHANGES_REQUESTED** [review]
Hi @Ferioney, thanks for updating the PR.  Just a few cosmetic changes are left to make now:
- have added a comment - please address
- please add your full name and surname with `@author` tag to the javadocs of the classes you've changed
- please add information about the new flag and feature to the documentation `spring-cloud-openfeign.adoc` in the `Feign Spring Cloud CircuitBreaker Support` section (after changing the docs rebuild with `-Pdocs` mvn flag to recreate auto-generated doc files).

File: spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/FeignCircuitBreakerTargeter.java (hunk 2 above)
Comment (Ferioney) [reply spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/FeignCircuitBreakerTargeter.java:87]: Removed

**Comment by Ferioney** [comment]
@OlgaMaciaszek 
- `this.` removed
- I added `@ author` tags to all changed classes (except for classes that don't have `@ author` tag)
- Documentation updated

**Review by OlgaMaciaszek — APPROVED** [review]
LGTM.

**Pull request merged**